Tests/Tests/Core/Utilities/DispatchTimer/DispatchTimerTests.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  DispatchTimerTests.swift
+//  PovioKit_Tests
+//
+//  Created by Borut Tomazin on 19/09/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import XCTest
+import PovioKit
+
+final class DispatchTimerTests: XCTestCase {
+  func test_dispatchTimer_instance_noRepeating() {
+    let timer = DispatchTimer()
+    XCTAssertFalse(timer.isActive)
+    
+    let promise = expectation(description: "Wait for timer...")
+    timer.schedule(interval: .milliseconds(100), repeating: false, on: .main) {
+      promise.fulfill()
+    }
+    
+    XCTAssertTrue(timer.isActive)
+    waitForExpectations(timeout: 0.5)
+    XCTAssertFalse(timer.isActive)
+  }
+  
+  func test_dispatchTimer_instance_repeating() {
+    let timer = DispatchTimer()
+    XCTAssertFalse(timer.isActive)
+    
+    let promise = expectation(description: "Wait for timer...")
+    promise.expectedFulfillmentCount = 5
+    
+    var repeatCount = 0
+    timer.schedule(interval: .milliseconds(50), repeating: true, on: .main) {
+      promise.fulfill()
+      repeatCount += 1
+      if repeatCount >= 5 {
+        timer.stop()
+      }
+    }
+    
+    XCTAssertTrue(timer.isActive)
+    waitForExpectations(timeout: 0.5)
+    XCTAssertFalse(timer.isActive)
+  }
+  
+  func test_dispatchTimer_static_noRepeating() {
+    let promise = expectation(description: "Wait for timer...")
```

Comment:
Maybe we could add `promise.expectedFulfillmentCount = 1` like in previous test to make it more assertive what we are trying to achieve.

Reply:
That feels redundant. We all know the default value is 1, so why overriding defaults 🤷‍♂️

Reply:
> That feels redundant. We all know the default value is 1, so why overriding defaults 🤷‍♂️

The way you tested UIViews with the preconditions is also a bit redundant though you did it because you felt that they provided value, that's how I saw them here as well. Simply a bit easier to read for someone looking at them for the first time.

Reply:
I agree, it is. But that's how I felt it should be 🤷‍♂️